ruth/simulator/singlenode.py

```diff
@@ -128,6 +127,24 @@ def simulate(
             step += 1
         logger.info(f"Simulation done in {self.sim.duration}.")
 
+    def compute_alternatives(self, alternatives_providers: List[AlternativesProvider], vehicles: List[Vehicle]):
```

Comment:
No need for this to be a method, it could just be a function. Makes it clearer that it doesn't depend on the simulator and doesn't use `self`.